### ISSUE TITLE
Class auth does not exist

### ISSUE THREAD

**Issue by GeoffreyFrogeye** [firefly-iii/firefly-iii#2305]
**Bug description**
I am running Firefly III from branch develop, and my problem is:
When I try to update my 4.7.17 installation [straight from Github](https://docs.firefly-iii.org/en/latest/installation/upgrading.html#straight-from-github), I get the following problem at this step:

```
$ composer install --no-dev
> @php -r "if (!(getenv('DYNO'))===false){file_exists('.env') || copy('.deploy/heroku/.env.heroku', '.env');copy('.deploy/heroku/.locales', '.locales');}"
Loading composer repositories with package information
Installing dependencies from lock file
Nothing to install or update
Generating optimized autoload files
> Illuminate\Foundation\ComposerScripts::postAutoloadDump
> @php artisan firefly:instructions install

In Container.php line 790:
                             
  Class auth does not exist  
```

Ignoring this and apptempting the following steps results in the same message but in color. Opening the browser at this points gives the same error.

**Steps to reproduce**

Luckily this can be reproduced on a fresh install:

```
git clone https://github.com/firefly-iii/firefly-iii.git
cd firefly-iii
git checkout develop
php artisan key:generate
composer install --no-scripts --no-dev
composer install --no-dev
```

Had no luck trying the `composer create-project` method since I couldn't get it to use the `develop` branch (always pulled 4.7.17).

**Expected behavior**
```
+------------------------------------------------------------------------------+
|                                                                              |
| Thank you for installing Firefly III, v4.8.0!                                |
|                                                                              |
|                                                                              |
+------------------------------------------------------------------------------+
```

**Extra info**

The `/debug` endpoint doesn't work with the bad commits so it is from the last good commit.

Debug information generated at 2019-06-08 21:27:38 Europe/Paris for Firefly III version **4.8.0**.

| Variable | Content |
| --- | --- |
| FF version | 4.8.0 |
| FF API version | 0.10.0 |
| App environment | local |
| App debug mode | true |
| App cache driver | file |
| App logging | , daily |
| PHP version | 7.3.6 |
| Display errors | Off |
| Session start | 2019-06-01 00:00:00 |
| Session end | 2019-06-30 23:59:59 |
| Session first | 2015-05-26 00:00:00 |
| Error reporting | ALL errors |
| Host | Linux |
| Interface | fpm-fcgi |
| UserID | 1 |
| Attempt at "en" | false |
| Attempt at "English" | false |
| Attempt at "en_US.utf8" | 'en_US.utf8' |
| Attempt at "en_US.UTF-8" | 'en_US.UTF-8' |
| DB drivers | mysql, pgsql |
| Current driver | pgsql |
| Login provider |   |
| Storage disks | local-upload  |
| Using Sandstorm? | no |
| Is Sandstorm (.env) | false |
| Is Docker (.env) | false |
| bunq uses sandbox | false |
| Trusted proxies (.env) |  |
| User agent | Mozilla/5.0 (X11; Linux x86_64; rv:67.0) Gecko/20100101 Firefox/67.0 |
| Loaded extensions | Core, date, libxml, openssl, pcre, zlib, ctype, dom, fileinfo, filter, hash, json, mbstring, pcntl, SPL, PDO, session, posix, readline, Reflection, standard, SimpleXML, Phar, tokenizer, xml, xmlreader, xmlwriter, mysqlnd, cgi-fcgi, bcmath, calendar, curl, exif, gd, gettext, iconv, imap, intl, ldap, mysqli, pdo_mysql, pdo_pgsql, pgsql, soap, sockets, zip, apcu, igbinary, imagick, redis, Zend OPcache |

**Stacktrace**
```
local.ERROR: Class auth does not exist {"exception":"[object] (ReflectionException(code: -1): Class auth does not exist at /opt/firefly-iii.new/vendor/laravel/framework/src/Illuminate/Container/Container.php:790)
[stacktrace]
#0 /opt/firefly-iii.new/vendor/laravel/framework/src/Illuminate/Container/Container.php(790): ReflectionClass->__construct('auth')
#1 /opt/firefly-iii.new/vendor/laravel/framework/src/Illuminate/Container/Container.php(667): Illuminate\\Container\\Container->build('auth')
#2 /opt/firefly-iii.new/vendor/laravel/framework/src/Illuminate/Container/Container.php(615): Illuminate\\Container\\Container->resolve('auth', Array)
#3 /opt/firefly-iii.new/vendor/laravel/framework/src/Illuminate/Foundation/Application.php(757): Illuminate\\Container\\Container->make('auth', Array)
#4 /opt/firefly-iii.new/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php(121): Illuminate\\Foundation\\Application->make('auth', Array)
#5 /opt/firefly-iii.new/vendor/laravel/framework/src/Illuminate/Foundation/helpers.php(162): app('Illuminate\\\\Cont...')
#6 /opt/firefly-iii.new/app/Exceptions/Handler.php(129): auth()
#7 /opt/firefly-iii.new/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(314): FireflyIII\\Exceptions\\Handler->report(Object(Symfony\\Component\\Debug\\Exception\\FatalThrowableError))
#8 /opt/firefly-iii.new/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(122): Illuminate\\Foundation\\Http\\Kernel->reportException(Object(Symfony\\Component\\Debug\\Exception\\FatalThrowableError))
#9 /opt/firefly-iii.new/public/index.php(76): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#10 {main}
"}
```

**Bissect results**
first bad commit: [779650f63dcaba09d50e54108161d94603457a5e] Deprecate the export function.

**Aknowledged issues**

- #347
- #332
- #758

Proposed workarrounds does not work or apply.

**Comment by JC5** [comment]
Don’t run the develop branch. The error may be caused by missing dependencies. It’s not caused by anything in the Firefly III code.

**Comment by GeoffreyFrogeye** [comment]
Oh ok. Which branch / installation method should I use to tinker with the new API and forms?

**Comment by JC5** [comment]
The develop branch should work.

```
git clone etc.
composer install --no-dev --no-scripts
cp .env.example .env
php artisan key:generate
composer install --no-dev
```

Once the DB is configured, you can run:

```
php artisan migrate:refresh --seed
php artisan firefly:upgrade-database
php artisan firefly:verify
php artisan passport:install
```

**Comment by GeoffreyFrogeye** [comment]
Sounds like it's fixed as from 85f9c256a1a40de4a396b167a4f5b5bb1a5dc49c. Thanks!

**Comment by JC5** [comment]
:+1:

Thanks for keeping me on my toes :wink:
